lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/GrokNamedGroupExtractor.java

```diff
@@ -50,28 +52,26 @@ private GrokUtils() {}
                     ")?" +
                     "\\}");
 
-    public static final Pattern NAMED_REGEX = Pattern
-            .compile("\\(\\?<([a-zA-Z][a-zA-Z0-9]*)>");
-
-    private static Set<String> getNameGroups(String regex)
+    public static Map<String, String> namedGroups(Matcher matcher, Boolean isCompiled)
     {
-        Set<String> namedGroups = new LinkedHashSet<String>();
-        Matcher m = NAMED_REGEX.matcher(regex);
-        while (m.find()) {
-            namedGroups.add(m.group(1));
+        if (isCompiled) {
+            Set<String> sortedKeys = matcher.namedGroups().entrySet().stream()
```

Comment:
Creating `sortedKeys` on each invocation is still going to be expensive and should be avoidable by extracting those values once and storing the result inside of `Grok